schema_utils_test.py

```diff
@@ -433,6 +433,55 @@ def test_schemas_are_correctly_validated(self):
             with self.assertRaises((AssertionError, KeyError)):
                 validate_schema(schema)
 
+    def test_is_regex_raises_not_implemented_error(self):
+        """Test if is_regex(obj) raises NotImplementedError."""
+        with self.assertRaises(NotImplementedError):
+            schema_utils._Validators.is_regex('string') # pylint: disable=protected-access
+
+    def test_matches_regex_raises_not_implemented_error(self):
+        """Test if matches_regex(obj) raises NotImplementedError."""
+        with self.assertRaises(NotImplementedError):
+            schema_utils._Validators.matches_regex('s1', 's2') # pylint: disable=protected-access
+
+    def test_normalize_against_schema_raises_exception(self):
+        """Tests if normalize against schema raises exception
+        for invalid key.
+        """
+        with self.assertRaises(Exception):
+            schema = {SCHEMA_KEY_TYPE: 'invalid'}
+            schema_utils.normalize_against_schema('obj', schema)
+
+    def test_is_nonempty(self):
+        """Tests if static method is_nonempty returns true iff obj
+        is not empty str.
+        """
+        is_nonempty = schema_utils._Validators.get('is_nonempty') # pylint: disable=protected-access
+        self.assertTrue(is_nonempty('non-empty string'))
+        self.assertFalse(is_nonempty(''))
+
+    def test_is_at_most(self):
+        """Tests if static method is_at_most returns true iff obj
+        is at most a value.
+        """
+        is_at_most = schema_utils._Validators.get('is_at_most') # pylint: disable=protected-access
+        self.assertTrue(is_at_most(2, 3))
```

Comment:
Here and below, add tests for boundary cases. E.g. in this case "is at most 2", in the following case "has len at least 1".